types/blueimp-load-image/index.d.ts

```diff
@@ -1,12 +1,16 @@
-// Type definitions for blueimp-load-image 2.23
+// Type definitions for blueimp-load-image 5.14
 // Project: https://github.com/blueimp/JavaScript-Load-Image
 // Definitions by: Evan Kesten <https://github.com/ebk46>
 //                 Konstantin Lukaschenko <https://github.com/KonstantinLukaschenko>
 //                 Saeid Rezaei <https://github.com/moeinio>
+//                 Zak Barbuto <https://github.com/zbarbuto>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 declare namespace loadImage {
     type LoadImageCallback = (eventOrImage: Event | HTMLCanvasElement | HTMLImageElement, data?: MetaData) => void;
+    type LoadImageResult = MetaData & {
```

Comment:
I'm not familiar with this particular API, so please bear with me.  The pattern I usually see is that the callback variant accepts a callback with a `T` parameter and the promise variant returns a `Promise<T>`.  In this case, the callback and promise types look unrelated?  And the callback variant returns a value (rather than being void)?

Reply:
I agree that `callback: (result: T) => void` and `Promise<T>` is typically what you see - but in the case of this library (and I've only got the docs to go by here) it seems to be a bit of a quirk that the callback takes two arguments (image and the result metadata) and the promise returns the result as a single payload:

![image](https://user-images.githubusercontent.com/9100419/100943967-2f2afe00-354e-11eb-92d0-9c1e4d129bdd.png)

![image](https://user-images.githubusercontent.com/9100419/100943986-38b46600-354e-11eb-9297-43d1ac1d9ab5.png)

![image](https://user-images.githubusercontent.com/9100419/100943993-3e11b080-354e-11eb-91c2-6c332ff8bac1.png)

Reply:
That's weird.  Thanks for doing the due diligence!  It's still not clear that callback version should have a non-void return type, but I'm willing to say that's out of scope for your change.